persistence/src/main/scala/com/lightbend/lagom/internal/persistence/cluster/ClusterDistribution.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.persistence.cluster
+
+import java.net.URLEncoder
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{ Actor, ActorRef, ActorSystem, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider, Props, Terminated }
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.{ ClusterSharding, ClusterShardingSettings, ShardRegion }
+import com.lightbend.lagom.internal.persistence.cluster.ClusterDistribution.EnsureActive
+import com.typesafe.config.Config
+
+import scala.concurrent.duration._
+
+/**
+ * Settings for cluster distribution.
+ *
+ * @param clusterShardingSettings The cluster sharding settings.
+ * @param ensureActiveInterval The interval at which entities are ensured to be active.
+ */
+case class ClusterDistributionSettings(
+  clusterShardingSettings: ClusterShardingSettings,
+  ensureActiveInterval:    FiniteDuration
+)
+
+object ClusterDistributionSettings {
+  def apply(system: ActorSystem): ClusterDistributionSettings = {
+    val clusterShardingSettings = ClusterShardingSettings(system)
+    val config = system.settings.config.getConfig("akka.cluster.distribution")
+    ClusterDistributionSettings(config, clusterShardingSettings)
+  }
+
+  def apply(config: Config, clusterShardingSettings: ClusterShardingSettings): ClusterDistributionSettings = {
+    val ensureActiveInterval = config.getDuration("ensure-active-interval", TimeUnit.MILLISECONDS).milliseconds
+    ClusterDistributionSettings(clusterShardingSettings, ensureActiveInterval)
+  }
+}
+
+object ClusterDistribution extends ExtensionId[ClusterDistribution] with ExtensionIdProvider {
+
+  override def createExtension(system: ExtendedActorSystem): ClusterDistribution =
+    new ClusterDistribution(system)
+
+  override def lookup = ClusterDistribution
+
+  /**
+   * Sent to each entity in a cluster distribution to ensure it's active.
+   */
+  case class EnsureActive(entityId: EntityId)
```

Comment:
Won't `Identify` get handled by the shard region actor then though?  ie, it will be responded to directly by the shard region actor, rather than delivered to its receive, and so won't trigger the creation of the shard?